src/table/hooks/use-head-icons.tsx

```diff
@@ -0,0 +1,15 @@
+import React, { useMemo } from 'react';
+import Lock from '@qlik-trial/sprout/icons/react/Lock';
+import Descending from '@qlik-trial/sprout/icons/react/Descending';
+import Ascending from '@qlik-trial/sprout/icons/react/Ascending';
+import { Column } from '../../types';
+
+const useHeadIcons = (column: Column) => {
```

Comment:
unsure if this should be a hook or not. no need to memo really, but it uses components. It is mostly to get the logic out of the component

Reply:
There is some potential to wrap it in `useMemo` as the return value is a non-primitive value. Which would always have triggered a re-render of the `StyledSortButton` component. But if that component is not wrapped in `React.memo` that will occur anyway.

By the looks of it `StyledSortButton` is not using `React.memo` so the `useMemo` call should not be needed. BUT might be that the `styled` function is memoizing the component but I don't really know.

Reply:
ok yeah no real need to memo anything here. Anytime `HeadCellContent ` rerenders it should update the icons. So this file should just should just be named `get-head-icons.tsx`?

Reply:
Yeah I think you can. It could be useful for the virtual scroll table but might as well wait and see with that

Reply:
fixed